digitalfabrik/entitlementcard
# 1430: Extend card add params

### Short description

As a user i want the forms in the self service portal to be prefilled with my personal information if i click "Karte verlängern" with an existing card

#### Note
I'm not super happy with this solution, but since we can not generate proper functions from the `frontend/build-configs/index` to the `buildConfig.dart` to provide a function for each project to generate the applicationUrl with (or without) queryParams i didn't find a better solution. 

I mean we could maybe adjust the build config generation to be able to generate functions with parameters but this means lots of effort.

At least this solution avoids multiple linting ignore that would be needed if we provide a query param key object instead of single string values that can be null. Feel free to suggest a better solution but beware that you should run the buildConfig for all the projects to ensure that nothing breaks.

### Proposed changes



- pass query params keys to the app, ensure that they are the same as in administration
- provide a function that return the applicationUrl with queryParams
- open the self service portal with queryParams if a card exists
- add tests
- workaround for #1802 

### Side effects



- may create a wrong applicationUrl for other projects. so open "Beantragen" for other projects

### Testing
- Create a card for koblenz that is less than 90 days valid from now
- Activate this card
- Change the end date in "UserEntitlements" to a later date
- Click on the "Karte verlängern" Button or go to more actions and extend the card there
- All form fields should be prefilled with valid information
- Note: I also tested "Umlaute and dots" and everything worked fine. Please also test that properly



### Resolved issues



partially fixes: #1430
